src/services/matrix.rs

```diff
@@ -1239,9 +1239,9 @@ pub mod matrix {
         }
     }
     pub async fn register(
-        homeserver: String,
-        username: String,
-        password: String,
+        homeserver: &String,
```

Comment:
Just like previous suggestion, everywhere where you receive a `String` or `&String` it's usually better to receive a `&str`(or in few cases `Rc<String>`.
```suggestion
        homeserver: &str,
```